src/ergw_aaa_diameter_srv.erl

```diff
@@ -264,28 +264,16 @@ handle_cast(stop, State) ->
 
 handle_info({'DOWN', MRef, _Type, Pid, Info}, #state{peers = Peers0} = State)
   when is_map_key(MRef, Peers0) ->
-    ?LOG(alert, "got down for pending request ~0p (~p) with ~p",
-	 [maps:get(MRef, Peers0), Pid, Info]),
+    ?LOG(alert, "Diameter request process terminated unexpected with ~0tp (req: ~0tp, pid: ~0p)",
+	 [Info, maps:get(MRef, Peers0), Pid]),
 
     {_, _, PeerRef, Caps} = maps:get(MRef, Peers0),
-    {Result, Peers} = release_peer(PeerRef, Caps, maps:without([Pid, MRef], Peers0)),
-    case Result of
-	{error, Error} ->
-	    ?LOG(critical, "release peer failed with ~0p", [Error]),
-	    ct:fail("release peer failed with ~0p", [Error]),
-	    ok;
-	ok ->
-	    ok
-    end,
+    Peers = release_peer(PeerRef, Caps, maps:without([Pid, MRef], Peers0)),
     ets:match_delete(?MODULE, {{'_', Pid}, MRef}),
     {noreply, State#state{peers = Peers}};
 
 handle_info({'DOWN', MRef, _Type, Pid, _Info}, State) ->
     ets:match_delete(?MODULE, {{'_', Pid}, MRef}),
-    {noreply, State};
-
-handle_info(Info, State) ->
-    ?LOG(warning, "handle_info: ~p", [Info]),
     {noreply, State}.
 
```

Comment:
It used to be recommended in a gen_server to have a cluse handle_info of unknown content. Perhaps it is no longer a good idea.

Reply:
Production code should IMHO not get call/cast/info message that are not handled. Sure, it is always possible to send something though a remote shell, but I would that not call a **normal** situation.

Reply:
@javiermtorres @ebengt I would be ok with a policy to always add catch all clauses for those with logs at level `alert`  (or even `emergency`). That would make them noisy enough to catch them, but still be somewhat safe.

Very high log levels are IMHO justified because missing a info/cast/call could mean that something important was missed and it is impossible so determine if it was important without understanding the message.

Reply:
Ok, I'd be fine with a high log level. The alternative is a crash (if we omit the catchall clause) that is equally visible.

Since infos catch any sent message, it's easier to stumble upon issues of processes sending messages to the wrong pid than calls/casts. I'd agree to not using catchall clauses for calls/casts.